src/js/angular/ttyg/controllers/ttyg-view.controller.js

```diff
@@ -276,6 +279,49 @@ function TTYGViewCtrl($rootScope, $scope, $http, $timeout, $translate, $uibModal
             });
     };
 
+    $scope.onCloneAgent = (agent) => {
+        let agentToClone = agent;
+        if (!agentToClone) {
+            agentToClone = TTYGContextService.getSelectedAgent();
+        }
+        const agentFormModel = agentFormModelMapper(agentToClone);
+        agentFormModel.name = `clone-${agentFormModel.name}`;
+        const activeRepositoryInfo = repositoryInfoMapper($repositories.getActiveRepositoryObject());
+        console.log(`clone agent`, agentFormModel, agent);
+        const options = {
+            templateUrl: 'js/angular/ttyg/templates/modal/agent-settings-modal.html',
+            controller: 'AgentSettingsModalController',
+            windowClass: 'agent-settings-modal',
+            resolve: {
+                dialogModel: function () {
+                    return new AgentSettingsModal(
+                        activeRepositoryInfo,
+                        $scope.activeRepositoryList,
+                        agentFormModel,
+                        AGENT_OPERATION.CLONE
+                    );
+                }
+            },
+            size: 'lg'
+        };
+        $uibModal.open(options).result.then(
+            // confirmed handler
+            (data) => {
+                TTYGService.createAgent(data)
+                    .then((updatedAgent) => {
+                        toastr.success($translate.instant("ttyg.agent.messages.agent_save_successfully", {agentName: updatedAgent.name}));
+                        const hasSelectedAgent = TTYGContextService.getSelectedAgent();
+                        if (hasSelectedAgent && data.id === hasSelectedAgent.id) {
```

Comment:
Could this actually happen?
Is the id not unique?

Reply:
It is. This is also not needed for the clone action.

Reply:
removed